src/common/interceptors/response.interceptors.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  HttpStatus,
+  HttpException,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { SuccessResponseDto, ErrorResponseDto } from '../dto/response.dto';
+
+@Injectable()
+export class ResponseInterceptor<T>
+  implements NestInterceptor<T, SuccessResponseDto<T>>
+{
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<SuccessResponseDto<T>> {
+    return next.handle().pipe(
+      map((data) => ({
+        statusCode: HttpStatus.OK,
+        data,
+      })),
+    );
+  }
+}
+
+@Injectable()
+export class ErrorInterceptor
+  implements NestInterceptor<Error, ErrorResponseDto>
+{
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<ErrorResponseDto> {
+    return next.handle().pipe(
+      map((error) => {
+        const response = context.switchToHttp().getResponse();
```

Comment:
I do not see any use of response variable.
Could you explain What it does and why do we need it?